### PR TITLE
adding links to elaborative testing

### LINKED ISSUE #36: Device better ways to link examples to an op specification
### Request description

Currently, the spec inline examples, demonstrating the working of an op, as follows:

```c
// %x: [[1, 2], [3, 4]]
// %y: [[5, 6], [7, 8]]
%z = stablehlo.add %x, %y : tensor<2x2xf32>
// %z: [[6, 8], [10, 12]]
```

In the near future, let's think about the best way of providing these examples. For example, can we just link to the interpreter implementation of the op, so that people can check out the test suite? Or, more generally, let's rename this section to "Implementation" and link to the source code?

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by sdasgup3**
fixes #36 

The indent is to differentiate the added link to the following "Back to Ops" link.

### CODE REVIEW

**Review by ghpvnist — APPROVED** [review]

File: docs/spec_draft.md [hunk 1]
```diff
@@ -229,6 +229,8 @@ the IEEE-754 specification.
 // %result: [[6, 8], [10, 12]]
 ```
 
+&nbsp;[More Examples](https://github.com/sdasgup3/stablehlo/blob/main/stablehlo/tests/interpret_add.mlir)
```
Comment (burmako) [inline docs/spec_draft.md:232]: Looks like this link is pointing to your fork of the repo :) How about using a relative link instead? I think `stablehlo/tests/interpret_add.mlir` will work correctly.

File: docs/spec_draft.md [hunk 2]
```diff
@@ -229,6 +229,8 @@ the IEEE-754 specification.
...
+
```
Comment (burmako) [inline docs/spec_draft.md:233]: Also what do you think about turning "Examples` into a link? We don't really need to keep examples in the markdown file once we have exhaustive tests, right?

**Review by burmako — CHANGES_REQUESTED** [review]

File: docs/spec_draft.md (hunk 2 above)
Comment (sdasgup3) [reply docs/spec_draft.md:235]: We already thought about this and here are our thoughts:

We want to keep a short example right in the spec to demonstrate, using a trivial (no corner cases) example, the functioning of the op and ask the reader to follow the link for more detailed tests. I believe having that quick-example-view provides better experience.  Having said that, I agree that the example section of spec should be concise: One representative should be enough to capture the high level behavior unless the there are multiple behaviors like bitwise ops.

What do you think?

Comment (burmako) [reply docs/spec_draft.md:235]: " believe having that quick-example-view provides better experience". Sounds good, let's give this a try and see how it works out.

**Review by burmako — APPROVED** [review]
